valueonly/context_test.go

```diff
@@ -0,0 +1,59 @@
+package valueonly
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"gotest.tools/v3/assert"
+)
+
+func TestContextValues(t *testing.T) {
+	ctx := context.WithValue(context.Background(), "testkey", "testvalue")
+
+	derivedCtx := &Context{ctx}
+	assert.Equal(t, derivedCtx.Value("testkey"), "testvalue")
+}
+
+func TestContextDeadline(t *testing.T) {
+	// Creating original context and deadlining it.
+	ctx := context.Background()
+	setDeadline := time.Now().Add(-time.Second)
+	ctx, cancel := context.WithDeadline(ctx, setDeadline)
+	defer cancel()
+
+	done := false
+	select {
+	case <-ctx.Done():
+		done = true
+	default:
+		done = false
+	}
+	actualDeadline, deadlineSet := ctx.Deadline()
+	assert.Equal(t, actualDeadline, setDeadline, "original context's deadline is set")
```

Comment:
General note, please try and use soft assertions, rather than hard ones, i.e.:
`assert.Check(t, cmp.Equal(x, y))`

Reply:
I understand - I found https://pkg.go.dev/gotest.tools/v3/assert#hdr-Assert_and_Check

I'll change the assertions if we decide to go forward rather than going with Shaun's solution, which looks promising.